### PR TITLE
make Close methods for the querier safe to call more than once.

### PR DESCRIPTION
**Pull request by krasi-georgiev**
from: https://github.com/prometheus/prometheus/issues/5408

`querier.Close` is called more than once somewhere in the rules Manager. I will still troubleshoot why this happens.

Regardless of this the method should be safe to be called more than once and this PR fixes this and also adds tests for the WAL and querier Close methods.

Signed-off-by: Krasi Georgiev <kgeorgie@redhat.com>

### CODE REVIEW

**Comment by bwplotka** [comment]
Unless it is urgent fix, but hopefully we can find the root cause, let me know, can try to debug this with you.

**Comment by krasi-georgiev** [comment]
Yes I will continue looking for the cause, but regardless, calling any public API should not cause a panic on a library.

File: querier.go [hunk 1]
```diff
@@ -253,12 +255,14 @@ func (q *blockQuerier) LabelValuesFor(string, labels.Label) ([]string, error) {
 
 func (q *blockQuerier) Close() error {
 	var merr tsdb_errors.MultiError
-
-	merr.Add(q.index.Close())
-	merr.Add(q.chunks.Close())
-	merr.Add(q.tombstones.Close())
-
-	return merr.Err()
+	if !q.closed {
```
Comment (bwplotka) [inline querier.go]: Usually in Golang you error out things first to have larger code blocks in "normal" flow. Can we fix that here too? Also `merr` variable should be definited just before usage.

Comment (krasi-georgiev) [reply querier.go]: updated, thanks!

**Pull request merged**